bug(barcode-scanning): `scan failed` on some Android devices

### Plugin(s)

- [X] Barcode Scanning
- [ ] Translation

### Did you test the latest version?

- [X] I use the latest version

### Platform(s)

- [X] Android
- [ ] iOS
- [ ] Web

### Current behavior

When calling the `scan(...)` the following error message appears in Android Studio:

```
scan failed.
com.google.mlkit.common.MlKitException: Failed to scan code.
at com.google.mlkit.vision.codescanner.internal.zze.zzc(com.google.android.gms:play-services-code-scanner@@16.0.0-beta3:4)
at com.google.mlkit.vision.codescanner.internal.zzf.onActivityResult(com.google.android.gms:play-services-code-scanner@@16.0.0-beta3:5)
at androidx.activity.result.ActivityResultRegistry.doDispatch(ActivityResultRegistry.java:409)
at androidx.activity.result.ActivityResultRegistry.dispatchResult(ActivityResultRegistry.java:366)
at androidx.activity.ComponentActivity.onActivityResult(ComponentActivity.java:712)
at android.app.Activity.dispatchActivityResult(Activity.java:8951)
at android.app.ActivityThread.deliverResults(ActivityThread.java:5987)
at android.app.ActivityThread.handleSendResult(ActivityThread.java:6033)
at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2574)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loopOnce(Looper.java:226)
at android.os.Looper.loop(Looper.java:313)
at android.app.ActivityThread.main(ActivityThread.java:8757)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

### Expected behavior

No exception.

### Reproduction

-

### Steps to reproduce

1. Call the `scan(...)` method on Android

Seems to be reproducible only on some devices (e.g. Samsung S21 with Android 13).

### Other information

_No response_

### Capacitor doctor

-

### Before submitting

- [X] I understand that incomplete issues (e.g. without reproduction) are closed.

@robingenz  New error:

``` chunk-vendors.19d1a91d.js:33987 Error: The Google Barcode Scanner Module is not available. You must install it first.
    at returnResult (VM3:772:32)
    at win.androidBridge.onmessage (VM3:747:21)
logError @ chunk-vendors.19d1a91d.js:33987
```

--- a/.changeset/popular-rings-fetch.md
+++ b/.changeset/popular-rings-fetch.md
@@ -1,0 +1,5 @@
+---
+'@capacitor-mlkit/barcode-scanning': patch
+---
+
+fix(android): `scan failed` on some devices

--- a/packages/barcode-scanning/README.md
+++ b/packages/barcode-scanning/README.md
@@ -487,6 +487,10 @@ installGoogleBarcodeScannerModule() => Promise<void>
 
 Install the Google <a href="#barcode">Barcode</a> Scanner module.
 
+**Attention**: This only starts the installation.
+The `googleBarcodeScannerModuleInstallProgress` event listener will
+notify you when the installation is complete.
+
 Only available on Android.
 
 **Since:** 5.1.0

--- a/packages/barcode-scanning/README.md
+++ b/packages/barcode-scanning/README.md
@@ -178,6 +178,16 @@ const openSettings = async () => {
   await BarcodeScanner.openSettings();
 };
 
+const isGoogleBarcodeScannerModuleAvailable = async () => {
+  const { available } =
+    await BarcodeScanner.isGoogleBarcodeScannerModuleAvailable();
+  return available;
+};
+
+const installGoogleBarcodeScannerModule = async () => {
+  await BarcodeScanner.installGoogleBarcodeScannerModule();
+};
+
 const checkPermissions = async () => {
   const { camera } = await BarcodeScanner.checkPermissions();
   return camera;
@@ -232,28 +242,28 @@ If you can't see the camera view, make sure all elements in the DOM are not visi
 
 <docgen-index>
 
-- [`startScan(...)`](#startscan)
-- [`stopScan()`](#stopscan)
-- [`readBarcodesFromImage(...)`](#readbarcodesfromimage)
-- [`scan(...)`](#scan)
-- [`isSupported()`](#issupported)
-- [`enableTorch()`](#enabletorch)
-- [`disableTorch()`](#disabletorch)
-- [`toggleTorch()`](#toggletorch)
-- [`isTorchEnabled()`](#istorchenabled)
-- [`isTorchAvailable()`](#istorchavailable)
-- [`openSettings()`](#opensettings)
-- [`isGoogleBarcodeScannerModuleAvailable()`](#isgooglebarcodescannermoduleavailable)
-- [`installGoogleBarcodeScannerModule()`](#installgooglebarcodescannermodule)
-- [`checkPermissions()`](#checkpermissions)
-- [`requestPermissions()`](#requestpermissions)
-- [`addListener('barcodeScanned', ...)`](#addlistenerbarcodescanned)
-- [`addListener('scanError', ...)`](#addlistenerscanerror)
-- [`addListener('googleBarcodeScannerModuleInstallProgress', ...)`](#addlistenergooglebarcodescannermoduleinstallprogress)
-- [`removeAllListeners()`](#removealllisteners)
-- [Interfaces](#interfaces)
-- [Type Aliases](#type-aliases)
-- [Enums](#enums)
+* [`startScan(...)`](#startscan)
+* [`stopScan()`](#stopscan)
+* [`readBarcodesFromImage(...)`](#readbarcodesfromimage)
+* [`scan(...)`](#scan)
+* [`isSupported()`](#issupported)
+* [`enableTorch()`](#enabletorch)
+* [`disableTorch()`](#disabletorch)
+* [`toggleTorch()`](#toggletorch)
+* [`isTorchEnabled()`](#istorchenabled)
+* [`isTorchAvailable()`](#istorchavailable)
+* [`openSettings()`](#opensettings)
+* [`isGoogleBarcodeScannerModuleAvailable()`](#isgooglebarcodescannermoduleavailable)
+* [`installGoogleBarcodeScannerModule()`](#installgooglebarcodescannermodule)
+* [`checkPermissions()`](#checkpermissions)
+* [`requestPermissions()`](#requestpermissions)
+* [`addListener('barcodeScanned', ...)`](#addlistenerbarcodescanned)
+* [`addListener('scanError', ...)`](#addlistenerscanerror)
+* [`addListener('googleBarcodeScannerModuleInstallProgress', ...)`](#addlistenergooglebarcodescannermoduleinstallprogress)
+* [`removeAllListeners()`](#removealllisteners)
+* [Interfaces](#interfaces)
+* [Type Aliases](#type-aliases)
+* [Enums](#enums)
 
 </docgen-index>
 
@@ -276,7 +286,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### stopScan()
 
@@ -290,7 +301,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### readBarcodesFromImage(...)
 
@@ -310,7 +322,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### scan(...)
 
@@ -323,7 +336,7 @@ Scan a barcode with a ready-to-use interface without WebView customization.
 On **Android**, this method is only available on devices with Google Play Services
 installed. Therefore, no camera permission is required.
 
-**Attention:** Before using this method on _Android_, first check if the Google <a href="#barcode">Barcode</a> Scanner module is available.
+**Attention:** Before using this method on *Android*, first check if the Google <a href="#barcode">Barcode</a> Scanner module is available.
 
 Only available on Android and iOS.
 
@@ -335,7 +348,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### isSupported()
 
@@ -351,7 +365,8 @@ Available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### enableTorch()
 
@@ -365,7 +380,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### disableTorch()
 
@@ -379,7 +395,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### toggleTorch()
 
@@ -393,7 +410,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### isTorchEnabled()
 
@@ -409,7 +427,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### isTorchAvailable()
 
@@ -425,7 +444,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### openSettings()
 
@@ -439,7 +459,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### isGoogleBarcodeScannerModuleAvailable()
 
@@ -455,7 +476,8 @@ Only available on Android.
 
 **Since:** 5.1.0
 
----
+--------------------
+
 
 ### installGoogleBarcodeScannerModule()
 
@@ -469,7 +491,8 @@ Only available on Android.
 
 **Since:** 5.1.0
 
----
+--------------------
+
 
 ### checkPermissions()
 
@@ -485,7 +508,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### requestPermissions()
 
@@ -501,7 +525,8 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### addListener('barcodeScanned', ...)
 
@@ -522,7 +547,8 @@ Available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### addListener('scanError', ...)
 
@@ -543,7 +569,8 @@ Available on Android and iOS.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### addListener('googleBarcodeScannerModuleInstallProgress', ...)
 
@@ -564,7 +591,8 @@ Available on Android.
 
 **Since:** 5.1.0
 
----
+--------------------
+
 
 ### removeAllListeners()
 
@@ -576,9 +604,11 @@ Remove all listeners for this plugin.
 
 **Since:** 0.0.1
 
----
+--------------------
+
 
 ### Interfaces
+
 
 #### StartScanOptions
 
@@ -587,11 +617,13 @@ Remove all listeners for this plugin.
 | **`formats`**    | <code>BarcodeFormat[]</code>                      | Improve the speed of the barcode scanner by configuring the barcode formats to scan for. | 0.0.1 |
 | **`lensFacing`** | <code><a href="#lensfacing">LensFacing</a></code> | Configure the camera (front or back) to use.                                             | 0.0.1 |
 
+
 #### ReadBarcodesFromImageResult
 
 | Prop           | Type                   | Description            | Since |
 | -------------- | ---------------------- | ---------------------- | ----- |
 | **`barcodes`** | <code>Barcode[]</code> | The detected barcodes. | 0.0.1 |
+
 
 #### Barcode
 
@@ -604,6 +636,7 @@ Remove all listeners for this plugin.
 | **`rawValue`**     | <code>string</code>                                                                   | The barcode value in a machine readable format.                                                                                                            | 0.0.1 |
 | **`valueType`**    | <code><a href="#barcodevaluetype">BarcodeValueType</a></code>                         | The barcode value type.                                                                                                                                    | 0.0.1 |
 
+
 #### ReadBarcodesFromImageOptions
 
 | Prop          | Type                         | Description                                                                              | Since |
@@ -611,11 +644,13 @@ Remove all listeners for this plugin.
 | **`formats`** | <code>BarcodeFormat[]</code> | Improve the speed of the barcode scanner by configuring the barcode formats to scan for. | 0.0.1 |
 | **`path`**    | <code>string</code>          | The local path to the image file.                                                        | 0.0.1 |
 
+
 #### ScanResult
 
 | Prop           | Type                   | Description            | Since |
 | -------------- | ---------------------- | ---------------------- | ----- |
 | **`barcodes`** | <code>Barcode[]</code> | The detected barcodes. | 0.0.1 |
+
 
 #### ScanOptions
 
@@ -623,11 +658,13 @@ Remove all listeners for this plugin.
 | ------------- | ---------------------------- | ---------------------------------------------------------------------------------------- | ----- |
 | **`formats`** | <code>BarcodeFormat[]</code> | Improve the speed of the barcode scanner by configuring the barcode formats to scan for. | 0.0.1 |
 
+
 #### IsSupportedResult
 
 | Prop            | Type                 | Description                                      | Since |
 | --------------- | -------------------- | ------------------------------------------------ | ----- |
 | **`supported`** | <code>boolean</code> | Whether or not the barcode scanner is supported. | 0.0.1 |
+
 
 #### IsTorchEnabledResult
 
@@ -635,11 +672,13 @@ Remove all listeners for this plugin.
 | ------------- | -------------------- | ------------------------------------ | ----- |
 | **`enabled`** | <code>boolean</code> | Whether or not the torch is enabled. | 0.0.1 |
 
+
 #### IsTorchAvailableResult
 
 | Prop            | Type                 | Description                            | Since |
 | --------------- | -------------------- | -------------------------------------- | ----- |
 | **`available`** | <code>boolean</code> | Whether or not the torch is available. | 0.0.1 |
+
 
 #### IsGoogleBarcodeScannerModuleAvailableResult
 
@@ -647,11 +686,13 @@ Remove all listeners for this plugin.
 | --------------- | -------------------- | ------------------------------------------------------------------------------------- | ----- |
 | **`available`** | <code>boolean</code> | Whether or not the Google <a href="#barcode">Barcode</a> Scanner module is available. | 5.1.0 |
 
+
 #### PermissionStatus
 
 | Prop         | Type                                                                    | Since |
 | ------------ | ----------------------------------------------------------------------- | ----- |
 | **`camera`** | <code><a href="#camerapermissionstate">CameraPermissionState</a></code> | 0.0.1 |
+
 
 #### PluginListenerHandle
 
@@ -659,17 +700,20 @@ Remove all listeners for this plugin.
 | ------------ | ----------------------------------------- |
 | **`remove`** | <code>() =&gt; Promise&lt;void&gt;</code> |
 
+
 #### BarcodeScannedEvent
 
 | Prop          | Type                                        | Description         | Since |
 | ------------- | ------------------------------------------- | ------------------- | ----- |
 | **`barcode`** | <code><a href="#barcode">Barcode</a></code> | A detected barcode. | 0.0.1 |
 
+
 #### ScanErrorEvent
 
 | Prop          | Type                | Description        | Since |
 | ------------- | ------------------- | ------------------ | ----- |
 | **`message`** | <code>string</code> | The error message. | 0.0.1 |
+
 
 #### GoogleBarcodeScannerModuleInstallProgressEvent
 
@@ -678,17 +722,22 @@ Remove all listeners for this plugin.
 | **`state`**    | <code><a href="#googlebarcodescannermoduleinstallstate">GoogleBarcodeScannerModuleInstallState</a></code> | The current state of the installation.                         | 5.1.0 |
 | **`progress`** | <code>number</code>                                                                                       | The progress of the installation in percent between 0 and 100. | 5.1.0 |
 
+
 ### Type Aliases
+
 
 #### CameraPermissionState
 
 <code><a href="#permissionstate">PermissionState</a> | 'limited'</code>
 
+
 #### PermissionState
 
 <code>'prompt' | 'prompt-with-rationale' | 'granted' | 'denied'</code>
 
+
 ### Enums
+
 
 #### BarcodeFormat
 
@@ -708,12 +757,14 @@ Remove all listeners for this plugin.
 | **`UpcA`**       | <code>'UPC_A'</code>       | Only available on Android and iOS. | 0.0.1 |
 | **`UpcE`**       | <code>'UPC_E'</code>       | Only available on Android and iOS. | 0.0.1 |
 
+
 #### LensFacing
 
 | Members     | Value                | Since |
 | ----------- | -------------------- | ----- |
 | **`Front`** | <code>'FRONT'</code> | 0.0.1 |
 | **`Back`**  | <code>'BACK'</code>  | 0.0.1 |
+
 
 #### BarcodeValueType
 
@@ -732,6 +783,7 @@ Remove all listeners for this plugin.
 | **`Url`**            | <code>'URL'</code>             | 0.0.1 |
 | **`Wifi`**           | <code>'WIFI'</code>            | 0.0.1 |
 | **`Unknown`**        | <code>'UNKNOWN'</code>         | 0.0.1 |
+
 
 #### GoogleBarcodeScannerModuleInstallState
 

--- a/packages/barcode-scanning/README.md
+++ b/packages/barcode-scanning/README.md
@@ -58,8 +58,8 @@ This plugin will use the following project variables (defined in your app’s `v
 - `$androidxCameraCoreVersion` version of `com.google.mlkit:barcode-scanning` (default: `1.1.0`)
 - `$androidxCameraLifecycleVersion` version of `com.google.mlkit:barcode-scanning` (default: `1.1.0`)
 - `$androidxCameraViewVersion` version of `com.google.mlkit:barcode-scanning` (default: `1.1.0`)
-- `$mlkitBarcodeScanningVersion` version of `com.google.mlkit:barcode-scanning` (default: `17.0.3`)
-- `$playServicesCodeScannerVersion` version of `com.google.mlkit:barcode-scanning` (default: `16.0.0-beta3`)
+- `$mlkitBarcodeScanningVersion` version of `com.google.mlkit:barcode-scanning` (default: `17.1.0`)
+- `$playServicesCodeScannerVersion` version of `com.google.mlkit:barcode-scanning` (default: `16.0.0`)
 
 ### iOS
 
@@ -232,25 +232,28 @@ If you can't see the camera view, make sure all elements in the DOM are not visi
 
 <docgen-index>
 
-* [`startScan(...)`](#startscan)
-* [`stopScan()`](#stopscan)
-* [`readBarcodesFromImage(...)`](#readbarcodesfromimage)
-* [`scan(...)`](#scan)
-* [`isSupported()`](#issupported)
-* [`enableTorch()`](#enabletorch)
-* [`disableTorch()`](#disabletorch)
-* [`toggleTorch()`](#toggletorch)
-* [`isTorchEnabled()`](#istorchenabled)
-* [`isTorchAvailable()`](#istorchavailable)
-* [`openSettings()`](#opensettings)
-* [`checkPermissions()`](#checkpermissions)
-* [`requestPermissions()`](#requestpermissions)
-* [`addListener('barcodeScanned', ...)`](#addlistenerbarcodescanned)
-* [`addListener('scanError', ...)`](#addlistenerscanerror)
-* [`removeAllListeners()`](#removealllisteners)
-* [Interfaces](#interfaces)
-* [Type Aliases](#type-aliases)
-* [Enums](#enums)
+- [`startScan(...)`](#startscan)
+- [`stopScan()`](#stopscan)
+- [`readBarcodesFromImage(...)`](#readbarcodesfromimage)
+- [`scan(...)`](#scan)
+- [`isSupported()`](#issupported)
+- [`enableTorch()`](#enabletorch)
+- [`disableTorch()`](#disabletorch)
+- [`toggleTorch()`](#toggletorch)
+- [`isTorchEnabled()`](#istorchenabled)
+- [`isTorchAvailable()`](#istorchavailable)
+- [`openSettings()`](#opensettings)
+- [`isGoogleBarcodeScannerModuleAvailable()`](#isgooglebarcodescannermoduleavailable)
+- [`installGoogleBarcodeScannerModule()`](#installgooglebarcodescannermodule)
+- [`checkPermissions()`](#checkpermissions)
+- [`requestPermissions()`](#requestpermissions)
+- [`addListener('barcodeScanned', ...)`](#addlistenerbarcodescanned)
+- [`addListener('scanError', ...)`](#addlistenerscanerror)
+- [`addListener('googleBarcodeScannerModuleInstallProgress', ...)`](#addlistenergooglebarcodescannermoduleinstallprogress)
+- [`removeAllListeners()`](#removealllisteners)
+- [Interfaces](#interfaces)
+- [Type Aliases](#type-aliases)
+- [Enums](#enums)
 
 </docgen-index>
 
@@ -273,8 +276,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### stopScan()
 
@@ -288,8 +290,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### readBarcodesFromImage(...)
 
@@ -309,8 +310,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### scan(...)
 
@@ -320,7 +320,10 @@ scan(options?: ScanOptions | undefined) => Promise<ScanResult>
 
 Scan a barcode with a ready-to-use interface without WebView customization.
 
-On **Android**, no camera permission is required.
+On **Android**, this method is only available on devices with Google Play Services
+installed. Therefore, no camera permission is required.
+
+**Attention:** Before using this method on _Android_, first check if the Google <a href="#barcode">Barcode</a> Scanner module is available.
 
 Only available on Android and iOS.
 
@@ -332,8 +335,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### isSupported()
 
@@ -349,8 +351,7 @@ Available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### enableTorch()
 
@@ -364,8 +365,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### disableTorch()
 
@@ -379,8 +379,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### toggleTorch()
 
@@ -394,8 +393,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### isTorchEnabled()
 
@@ -411,8 +409,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### isTorchAvailable()
 
@@ -428,8 +425,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### openSettings()
 
@@ -443,8 +439,37 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
+---
 
+### isGoogleBarcodeScannerModuleAvailable()
+
+```typescript
+isGoogleBarcodeScannerModuleAvailable() => Promise<IsGoogleBarcodeScannerModuleAvailableResult>
+```
+
+Check if the Google <a href="#barcode">Barcode</a> Scanner module is available.
+
+Only available on Android.
+
+**Returns:** <code>Promise&lt;<a href="#isgooglebarcodescannermoduleavailableresult">IsGoogleBarcodeScannerModuleAvailableResult</a>&gt;</code>
+
+**Since:** 5.1.0
+
+---
+
+### installGoogleBarcodeScannerModule()
+
+```typescript
+installGoogleBarcodeScannerModule() => Promise<void>
+```
+
+Install the Google <a href="#barcode">Barcode</a> Scanner module.
+
+Only available on Android.
+
+**Since:** 5.1.0
+
+---
 
 ### checkPermissions()
 
@@ -460,8 +485,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### requestPermissions()
 
@@ -477,8 +501,7 @@ Only available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### addListener('barcodeScanned', ...)
 
@@ -499,8 +522,7 @@ Available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### addListener('scanError', ...)
 
@@ -521,8 +543,28 @@ Available on Android and iOS.
 
 **Since:** 0.0.1
 
---------------------
+---
 
+### addListener('googleBarcodeScannerModuleInstallProgress', ...)
+
+```typescript
+addListener(eventName: 'googleBarcodeScannerModuleInstallProgress', listenerFunc: (event: GoogleBarcodeScannerModuleInstallProgressEvent) => void) => Promise<PluginListenerHandle> & PluginListenerHandle
+```
+
+Called when the Google <a href="#barcode">Barcode</a> Scanner module is installed.
+
+Available on Android.
+
+| Param              | Type                                                                                                                                          |
+| ------------------ | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| **`eventName`**    | <code>'googleBarcodeScannerModuleInstallProgress'</code>                                                                                      |
+| **`listenerFunc`** | <code>(event: <a href="#googlebarcodescannermoduleinstallprogressevent">GoogleBarcodeScannerModuleInstallProgressEvent</a>) =&gt; void</code> |
+
+**Returns:** <code>Promise&lt;<a href="#pluginlistenerhandle">PluginListenerHandle</a>&gt; & <a href="#pluginlistenerhandle">PluginListenerHandle</a></code>
+
+**Since:** 5.1.0
+
+---
 
 ### removeAllListeners()
 
@@ -534,11 +576,9 @@ Remove all listeners for this plugin.
 
 **Since:** 0.0.1
 
---------------------
-
+---
 
 ### Interfaces
-
 
 #### StartScanOptions
 
@@ -547,13 +587,11 @@ Remove all listeners for this plugin.
 | **`formats`**    | <code>BarcodeFormat[]</code>                      | Improve the speed of the barcode scanner by configuring the barcode formats to scan for. | 0.0.1 |
 | **`lensFacing`** | <code><a href="#lensfacing">LensFacing</a></code> | Configure the camera (front or back) to use.                                             | 0.0.1 |
 
-
 #### ReadBarcodesFromImageResult
 
 | Prop           | Type                   | Description            | Since |
 | -------------- | ---------------------- | ---------------------- | ----- |
 | **`barcodes`** | <code>Barcode[]</code> | The detected barcodes. | 0.0.1 |
-
 
 #### Barcode
 
@@ -566,7 +604,6 @@ Remove all listeners for this plugin.
 | **`rawValue`**     | <code>string</code>                                                                   | The barcode value in a machine readable format.                                                                                                            | 0.0.1 |
 | **`valueType`**    | <code><a href="#barcodevaluetype">BarcodeValueType</a></code>                         | The barcode value type.                                                                                                                                    | 0.0.1 |
 
-
 #### ReadBarcodesFromImageOptions
 
 | Prop          | Type                         | Description                                                                              | Since |
@@ -574,13 +611,11 @@ Remove all listeners for this plugin.
 | **`formats`** | <code>BarcodeFormat[]</code> | Improve the speed of the barcode scanner by configuring the barcode formats to scan for. | 0.0.1 |
 | **`path`**    | <code>string</code>          | The local path to the image file.                                                        | 0.0.1 |
 
-
 #### ScanResult
 
 | Prop           | Type                   | Description            | Since |
 | -------------- | ---------------------- | ---------------------- | ----- |
 | **`barcodes`** | <code>Barcode[]</code> | The detected barcodes. | 0.0.1 |
-
 
 #### ScanOptions
 
@@ -588,13 +623,11 @@ Remove all listeners for this plugin.
 | ------------- | ---------------------------- | ---------------------------------------------------------------------------------------- | ----- |
 | **`formats`** | <code>BarcodeFormat[]</code> | Improve the speed of the barcode scanner by configuring the barcode formats to scan for. | 0.0.1 |
 
-
 #### IsSupportedResult
 
 | Prop            | Type                 | Description                                      | Since |
 | --------------- | -------------------- | ------------------------------------------------ | ----- |
 | **`supported`** | <code>boolean</code> | Whether or not the barcode scanner is supported. | 0.0.1 |
-
 
 #### IsTorchEnabledResult
 
@@ -602,13 +635,17 @@ Remove all listeners for this plugin.
 | ------------- | -------------------- | ------------------------------------ | ----- |
 | **`enabled`** | <code>boolean</code> | Whether or not the torch is enabled. | 0.0.1 |
 
-
 #### IsTorchAvailableResult
 
 | Prop            | Type                 | Description                            | Since |
 | --------------- | -------------------- | -------------------------------------- | ----- |
 | **`available`** | <code>boolean</code> | Whether or not the torch is available. | 0.0.1 |
 
+#### IsGoogleBarcodeScannerModuleAvailableResult
+
+| Prop            | Type                 | Description                                                                           | Since |
+| --------------- | -------------------- | ------------------------------------------------------------------------------------- | ----- |
+| **`available`** | <code>boolean</code> | Whether or not the Google <a href="#barcode">Barcode</a> Scanner module is available. | 5.1.0 |
 
 #### PermissionStatus
 
@@ -616,13 +653,11 @@ Remove all listeners for this plugin.
 | ------------ | ----------------------------------------------------------------------- | ----- |
 | **`camera`** | <code><a href="#camerapermissionstate">CameraPermissionState</a></code> | 0.0.1 |
 
-
 #### PluginListenerHandle
 
 | Prop         | Type                                      |
 | ------------ | ----------------------------------------- |
 | **`remove`** | <code>() =&gt; Promise&lt;void&gt;</code> |
-
 
 #### BarcodeScannedEvent
 
@@ -630,29 +665,30 @@ Remove all listeners for this plugin.
 | ------------- | ------------------------------------------- | ------------------- | ----- |
 | **`barcode`** | <code><a href="#barcode">Barcode</a></code> | A detected barcode. | 0.0.1 |
 
-
 #### ScanErrorEvent
 
 | Prop          | Type                | Description        | Since |
 | ------------- | ------------------- | ------------------ | ----- |
 | **`message`** | <code>string</code> | The error message. | 0.0.1 |
 
+#### GoogleBarcodeScannerModuleInstallProgressEvent
+
+| Prop           | Type                                                                                                      | Description                                                    | Since |
+| -------------- | --------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------- | ----- |
+| **`state`**    | <code><a href="#googlebarcodescannermoduleinstallstate">GoogleBarcodeScannerModuleInstallState</a></code> | The current state of the installation.                         | 5.1.0 |
+| **`progress`** | <code>number</code>                                                                                       | The progress of the installation in percent between 0 and 100. | 5.1.0 |
 
 ### Type Aliases
-
 
 #### CameraPermissionState
 
 <code><a href="#permissionstate">PermissionState</a> | 'limited'</code>
 
-
 #### PermissionState
 
 <code>'prompt' | 'prompt-with-rationale' | 'granted' | 'denied'</code>
 
-
 ### Enums
-
 
 #### BarcodeFormat
 
@@ -672,14 +708,12 @@ Remove all listeners for this plugin.
 | **`UpcA`**       | <code>'UPC_A'</code>       | Only available on Android and iOS. | 0.0.1 |
 | **`UpcE`**       | <code>'UPC_E'</code>       | Only available on Android and iOS. | 0.0.1 |
 
-
 #### LensFacing
 
 | Members     | Value                | Since |
 | ----------- | -------------------- | ----- |
 | **`Front`** | <code>'FRONT'</code> | 0.0.1 |
 | **`Back`**  | <code>'BACK'</code>  | 0.0.1 |
-
 
 #### BarcodeValueType
 
@@ -698,6 +732,19 @@ Remove all listeners for this plugin.
 | **`Url`**            | <code>'URL'</code>             | 0.0.1 |
 | **`Wifi`**           | <code>'WIFI'</code>            | 0.0.1 |
 | **`Unknown`**        | <code>'UNKNOWN'</code>         | 0.0.1 |
+
+#### GoogleBarcodeScannerModuleInstallState
+
+| Members               | Value          | Since |
+| --------------------- | -------------- | ----- |
+| **`UNKNOWN`**         | <code>0</code> | 5.1.0 |
+| **`PENDING`**         | <code>1</code> | 5.1.0 |
+| **`DOWNLOADING`**     | <code>2</code> | 5.1.0 |
+| **`CANCELED`**        | <code>3</code> | 5.1.0 |
+| **`COMPLETED`**       | <code>4</code> | 5.1.0 |
+| **`FAILED`**          | <code>5</code> | 5.1.0 |
+| **`INSTALLING`**      | <code>6</code> | 5.1.0 |
+| **`DOWNLOAD_PAUSED`** | <code>7</code> | 5.1.0 |
 
 </docgen-api>
 

--- a/packages/barcode-scanning/android/build.gradle
+++ b/packages/barcode-scanning/android/build.gradle
@@ -7,8 +7,8 @@ ext {
     androidxCameraCoreVersion = project.hasProperty('androidxCameraCoreVersion') ? rootProject.ext.androidxCameraCoreVersion : '1.1.0'
     androidxCameraLifecycleVersion = project.hasProperty('androidxCameraLifecycleVersion') ? rootProject.ext.androidxCameraLifecycleVersion : '1.1.0'
     androidxCameraViewVersion = project.hasProperty('androidxCameraViewVersion') ? rootProject.ext.androidxCameraViewVersion : '1.1.0'
-    mlkitBarcodeScanningVersion = project.hasProperty('mlkitBarcodeScanningVersion') ? rootProject.ext.mlkitBarcodeScanningVersion : '17.0.3'
-    playServicesCodeScannerVersion = project.hasProperty('playServicesCodeScannerVersion') ? rootProject.ext.playServicesCodeScannerVersion : '16.0.0-beta3'
+    mlkitBarcodeScanningVersion = project.hasProperty('mlkitBarcodeScanningVersion') ? rootProject.ext.mlkitBarcodeScanningVersion : '17.1.0'
+    playServicesCodeScannerVersion = project.hasProperty('playServicesCodeScannerVersion') ? rootProject.ext.playServicesCodeScannerVersion : '16.0.0'
 }
 
 buildscript {

--- a/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/BarcodeScanner.java
+++ b/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/BarcodeScanner.java
@@ -33,6 +33,10 @@ import androidx.lifecycle.LifecycleOwner;
 import com.getcapacitor.Logger;
 import com.getcapacitor.PermissionState;
 import com.getcapacitor.PluginCall;
+import com.google.android.gms.common.moduleinstall.InstallStatusListener;
+import com.google.android.gms.common.moduleinstall.ModuleInstall;
+import com.google.android.gms.common.moduleinstall.ModuleInstallClient;
+import com.google.android.gms.common.moduleinstall.ModuleInstallRequest;
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.mlkit.vision.barcode.BarcodeScannerOptions;
 import com.google.mlkit.vision.barcode.BarcodeScanning;
@@ -156,6 +160,7 @@ public class BarcodeScanner implements ImageAnalysis.Analyzer {
     public void scan(ScanSettings scanSettings, ScanResultCallback callback) {
         GmsBarcodeScannerOptions options = buildGmsBarcodeScannerOptions(scanSettings);
         GmsBarcodeScanner scanner = GmsBarcodeScanning.getClient(plugin.getContext(), options);
+
         scanner
             .startScan()
             .addOnSuccessListener(
@@ -173,6 +178,38 @@ public class BarcodeScanner implements ImageAnalysis.Analyzer {
                     callback.error(exception);
                 }
             );
+    }
+
+    public void isGoogleCodeScannerModuleAvailable(IsGoogleCodeScannerModuleAvailableResultCallback callback) {
+        GmsBarcodeScanner scanner = GmsBarcodeScanning.getClient(plugin.getContext());
+        ModuleInstallClient moduleInstallClient = ModuleInstall.getClient(plugin.getContext());
+        moduleInstallClient.areModulesAvailable(scanner).addOnSuccessListener(response -> {
+            if (response.areModulesAvailable()) {
+                // TODO
+            }
+        }).addOnFailureListener(ex -> {
+            // TODO
+        });
+    }
+
+    public void installGoogleCodeScannerModule(InstallGoogleCodeScannerModuleResultCallback callback) {
+        GmsBarcodeScanner scanner = GmsBarcodeScanning.getClient(plugin.getContext());
+        InstallStatusListener listener = new ModuleInstallProgressListener();
+        ModuleInstallRequest moduleInstallRequest =
+                ModuleInstallRequest.newBuilder()
+                        .addApi(scanner)
+                        .setListener(listener)
+                        .build();
+        ModuleInstallClient moduleInstallClient = ModuleInstall.getClient(plugin.getContext());
+        moduleInstallClient
+                .installModules(moduleInstallRequest)
+                .addOnSuccessListener(moduleInstallResponse -> {
+                    if (moduleInstallResponse.areModulesAlreadyInstalled()) {
+                        // TODO
+                    }
+                }).addOnFailureListener(ex -> {
+                    // TODO
+                });
     }
 
     public boolean isSupported() {

--- a/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/BarcodeScanner.java
+++ b/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/BarcodeScanner.java
@@ -37,6 +37,7 @@ import com.google.android.gms.common.moduleinstall.InstallStatusListener;
 import com.google.android.gms.common.moduleinstall.ModuleInstall;
 import com.google.android.gms.common.moduleinstall.ModuleInstallClient;
 import com.google.android.gms.common.moduleinstall.ModuleInstallRequest;
+import com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate;
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.mlkit.vision.barcode.BarcodeScannerOptions;
 import com.google.mlkit.vision.barcode.BarcodeScanning;
@@ -48,7 +49,9 @@ import com.google.mlkit.vision.common.InputImage;
 
 public class BarcodeScanner implements ImageAnalysis.Analyzer {
 
+    @NonNull
     private final BarcodeScannerPlugin plugin;
+
     private final Point displaySize;
 
     @Nullable
@@ -65,6 +68,9 @@ public class BarcodeScanner implements ImageAnalysis.Analyzer {
 
     @Nullable
     private ScanSettings scanSettings;
+
+    @Nullable
+    private ModuleInstallProgressListener moduleInstallProgressListener;
 
     private boolean isTorchEnabled = false;
 
@@ -180,36 +186,45 @@ public class BarcodeScanner implements ImageAnalysis.Analyzer {
             );
     }
 
-    public void isGoogleCodeScannerModuleAvailable(IsGoogleCodeScannerModuleAvailableResultCallback callback) {
+    public void isGoogleBarcodeScannerModuleAvailable(IsGoogleBarodeScannerModuleAvailableResultCallback callback) {
         GmsBarcodeScanner scanner = GmsBarcodeScanning.getClient(plugin.getContext());
-        ModuleInstallClient moduleInstallClient = ModuleInstall.getClient(plugin.getContext());
-        moduleInstallClient.areModulesAvailable(scanner).addOnSuccessListener(response -> {
-            if (response.areModulesAvailable()) {
-                // TODO
-            }
-        }).addOnFailureListener(ex -> {
-            // TODO
-        });
-    }
-
-    public void installGoogleCodeScannerModule(InstallGoogleCodeScannerModuleResultCallback callback) {
-        GmsBarcodeScanner scanner = GmsBarcodeScanning.getClient(plugin.getContext());
-        InstallStatusListener listener = new ModuleInstallProgressListener();
-        ModuleInstallRequest moduleInstallRequest =
-                ModuleInstallRequest.newBuilder()
-                        .addApi(scanner)
-                        .setListener(listener)
-                        .build();
         ModuleInstallClient moduleInstallClient = ModuleInstall.getClient(plugin.getContext());
         moduleInstallClient
-                .installModules(moduleInstallRequest)
-                .addOnSuccessListener(moduleInstallResponse -> {
+            .areModulesAvailable(scanner)
+            .addOnSuccessListener(
+                response -> {
+                    boolean isAvailable = response.areModulesAvailable();
+                    callback.success(isAvailable);
+                }
+            )
+            .addOnFailureListener(
+                exception -> {
+                    callback.error(exception);
+                }
+            );
+    }
+
+    public void installGoogleBarcodeScannerModule(InstallGoogleBarcodeScannerModuleResultCallback callback) {
+        GmsBarcodeScanner scanner = GmsBarcodeScanning.getClient(plugin.getContext());
+        InstallStatusListener listener = new ModuleInstallProgressListener(this);
+        ModuleInstallRequest moduleInstallRequest = ModuleInstallRequest.newBuilder().addApi(scanner).setListener(listener).build();
+        ModuleInstallClient moduleInstallClient = ModuleInstall.getClient(plugin.getContext());
+        moduleInstallClient
+            .installModules(moduleInstallRequest)
+            .addOnSuccessListener(
+                moduleInstallResponse -> {
                     if (moduleInstallResponse.areModulesAlreadyInstalled()) {
-                        // TODO
+                        callback.error(new Exception(BarcodeScannerPlugin.ERROR_GOOGLE_BARCODE_SCANNER_MODULE_ALREADY_INSTALLED));
+                    } else {
+                        callback.success();
                     }
-                }).addOnFailureListener(ex -> {
-                    // TODO
-                });
+                }
+            )
+            .addOnFailureListener(
+                exception -> {
+                    callback.error(exception);
+                }
+            );
     }
 
     public boolean isSupported() {
@@ -309,6 +324,19 @@ public class BarcodeScanner implements ImageAnalysis.Analyzer {
                     image.close();
                 }
             );
+    }
+
+    public void handleGoogleBarcodeScannerModuleInstallProgress(
+        @ModuleInstallStatusUpdate.InstallState int state,
+        @Nullable Integer progress
+    ) {
+        plugin.notifyGoogleBarcodeScannerModuleInstallProgressListener(state, progress);
+        boolean isTerminateState = ModuleInstallProgressListener.isTerminateState(state);
+        if (isTerminateState && moduleInstallProgressListener != null) {
+            ModuleInstallClient moduleInstallClient = ModuleInstall.getClient(plugin.getContext());
+            moduleInstallClient.unregisterListener(moduleInstallProgressListener);
+            moduleInstallProgressListener = null;
+        }
     }
 
     private Point getDisplaySize() {

--- a/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/BarcodeScannerPlugin.java
+++ b/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/BarcodeScannerPlugin.java
@@ -8,6 +8,7 @@ import android.graphics.Point;
 import android.util.DisplayMetrics;
 import androidx.activity.result.ActivityResult;
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.camera.core.CameraSelector;
 import com.getcapacitor.JSArray;
 import com.getcapacitor.JSObject;
@@ -19,6 +20,7 @@ import com.getcapacitor.annotation.ActivityCallback;
 import com.getcapacitor.annotation.CapacitorPlugin;
 import com.getcapacitor.annotation.Permission;
 import com.getcapacitor.annotation.PermissionCallback;
+import com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate;
 import com.google.mlkit.vision.barcode.common.Barcode;
 import com.google.mlkit.vision.common.InputImage;
 import java.util.List;
@@ -34,9 +36,14 @@ public class BarcodeScannerPlugin extends Plugin {
 
     public static final String BARCODE_SCANNED_EVENT = "barcodeScanned";
     public static final String SCAN_ERROR_EVENT = "scanError";
+    public static final String GOOGLE_BARCODE_SCANNER_MODULE_INSTALL_PROGRESS_EVENT = "googleBarcodeScannerModuleInstallProgress";
     public static final String ERROR_SCAN_CANCELED = "scan canceled.";
     public static final String ERROR_PATH_MISSING = "path must be provided.";
     public static final String ERROR_LOAD_IMAGE_FAILED = "The image could not be loaded.";
+    public static final String ERROR_GOOGLE_BARCODE_SCANNER_MODULE_NOT_AVAILABLE =
+        "The Google Barcode Scanner Module is not available. You must install it first.";
+    public static final String ERROR_GOOGLE_BARCODE_SCANNER_MODULE_ALREADY_INSTALLED =
+        "The Google Barcode Scanner Module is already installed.";
     public static final String ERROR_PERMISSION_DENIED = "User denied access to camera.";
 
     private BarcodeScanner implementation;
@@ -166,34 +173,57 @@ public class BarcodeScannerPlugin extends Plugin {
             ScanSettings scanSettings = new ScanSettings();
             scanSettings.formats = formats;
 
-            implementation.scan(
-                scanSettings,
-                (
-                    new ScanResultCallback() {
-                        @Override
-                        public void success(Barcode barcode) {
-                            JSObject barcodeResult = BarcodeScannerHelper.createBarcodeResultForBarcode(barcode, null, null);
+            implementation.isGoogleBarcodeScannerModuleAvailable(
+                new IsGoogleBarodeScannerModuleAvailableResultCallback() {
+                    @Override
+                    public void success(boolean isAvailable) {
+                        if (isAvailable) {
+                            implementation.scan(
+                                scanSettings,
+                                (
+                                    new ScanResultCallback() {
+                                        @Override
+                                        public void success(Barcode barcode) {
+                                            JSObject barcodeResult = BarcodeScannerHelper.createBarcodeResultForBarcode(
+                                                barcode,
+                                                null,
+                                                null
+                                            );
 
-                            JSArray barcodeResults = new JSArray();
-                            barcodeResults.put(barcodeResult);
+                                            JSArray barcodeResults = new JSArray();
+                                            barcodeResults.put(barcodeResult);
 
-                            JSObject result = new JSObject();
-                            result.put("barcodes", barcodeResults);
-                            call.resolve(result);
-                        }
+                                            JSObject result = new JSObject();
+                                            result.put("barcodes", barcodeResults);
+                                            call.resolve(result);
+                                        }
 
-                        @Override
-                        public void cancel() {
-                            call.reject(ERROR_SCAN_CANCELED);
-                        }
+                                        @Override
+                                        public void cancel() {
+                                            call.reject(ERROR_SCAN_CANCELED);
+                                        }
 
-                        @Override
-                        public void error(Exception exception) {
-                            Logger.error("scan failed.", exception);
-                            call.reject(exception.getMessage());
+                                        @Override
+                                        public void error(Exception exception) {
+                                            String message = exception.getMessage();
+                                            Logger.error(message, exception);
+                                            call.reject(message);
+                                        }
+                                    }
+                                )
+                            );
+                        } else {
+                            call.reject(ERROR_GOOGLE_BARCODE_SCANNER_MODULE_NOT_AVAILABLE);
                         }
                     }
-                )
+
+                    @Override
+                    public void error(Exception exception) {
+                        String message = exception.getMessage();
+                        Logger.error(message, exception);
+                        call.reject(message);
+                    }
+                }
             );
         } catch (Exception exception) {
             String message = exception.getMessage();
@@ -288,6 +318,58 @@ public class BarcodeScannerPlugin extends Plugin {
         }
     }
 
+    @PluginMethod
+    public void isGoogleBarcodeScannerModuleAvailable(PluginCall call) {
+        try {
+            implementation.isGoogleBarcodeScannerModuleAvailable(
+                new IsGoogleBarodeScannerModuleAvailableResultCallback() {
+                    @Override
+                    public void success(boolean isAvailable) {
+                        JSObject result = new JSObject();
+                        result.put("available", isAvailable);
+                        call.resolve(result);
+                    }
+
+                    @Override
+                    public void error(Exception exception) {
+                        String message = exception.getMessage();
+                        Logger.error(message, exception);
+                        call.reject(message);
+                    }
+                }
+            );
+        } catch (Exception exception) {
+            String message = exception.getMessage();
+            Logger.error(message, exception);
+            call.reject(message);
+        }
+    }
+
+    @PluginMethod
+    public void installGoogleBarcodeScannerModule(PluginCall call) {
+        try {
+            implementation.installGoogleBarcodeScannerModule(
+                new InstallGoogleBarcodeScannerModuleResultCallback() {
+                    @Override
+                    public void success() {
+                        call.resolve();
+                    }
+
+                    @Override
+                    public void error(Exception exception) {
+                        String message = exception.getMessage();
+                        Logger.error(message, exception);
+                        call.reject(message);
+                    }
+                }
+            );
+        } catch (Exception exception) {
+            String message = exception.getMessage();
+            Logger.error(message, exception);
+            call.reject(message);
+        }
+    }
+
     @Override
     @PluginMethod
     public void requestPermissions(PluginCall call) {
@@ -343,6 +425,20 @@ public class BarcodeScannerPlugin extends Plugin {
             result.put("message", message);
 
             notifyListeners(SCAN_ERROR_EVENT, result);
+        } catch (Exception exception) {
+            Logger.error(exception.getMessage(), exception);
+        }
+    }
+
+    public void notifyGoogleBarcodeScannerModuleInstallProgressListener(int state, @Nullable Integer progress) {
+        try {
+            JSObject result = new JSObject();
+            result.put("state", state);
+            if (progress != null) {
+                result.put("progress", progress);
+            }
+
+            notifyListeners(GOOGLE_BARCODE_SCANNER_MODULE_INSTALL_PROGRESS_EVENT, result);
         } catch (Exception exception) {
             Logger.error(exception.getMessage(), exception);
         }

--- a/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/InstallGoogleBarcodeScannerModuleResultCallback.java
+++ b/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/InstallGoogleBarcodeScannerModuleResultCallback.java
@@ -1,0 +1,6 @@
+package io.capawesome.capacitorjs.plugins.mlkit.barcodescanning;
+
+public interface InstallGoogleBarcodeScannerModuleResultCallback {
+    void success();
+    void error(Exception exception);
+}

--- a/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/IsGoogleBarodeScannerModuleAvailableResultCallback.java
+++ b/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/IsGoogleBarodeScannerModuleAvailableResultCallback.java
@@ -1,0 +1,6 @@
+package io.capawesome.capacitorjs.plugins.mlkit.barcodescanning;
+
+public interface IsGoogleBarodeScannerModuleAvailableResultCallback {
+    void success(boolean isAvailable);
+    void error(Exception exception);
+}

--- a/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/ModuleInstallProgressListener.java
+++ b/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/ModuleInstallProgressListener.java
@@ -1,0 +1,33 @@
+package io.capawesome.capacitorjs.plugins.mlkit.barcodescanning;
+
+import static com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate.InstallState.STATE_CANCELED;
+import static com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate.InstallState.STATE_COMPLETED;
+import static com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate.InstallState.STATE_FAILED;
+
+import com.google.android.gms.common.moduleinstall.InstallStatusListener;
+import com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate;
+
+public class ModuleInstallProgressListener implements InstallStatusListener {
+    @Override
+    public void onInstallStatusUpdated(ModuleInstallStatusUpdate update) {
+        ModuleInstallStatusUpdate.ProgressInfo progressInfo = update.getProgressInfo();
+        // Progress info is only set when modules are in the progress of downloading.
+        if (progressInfo != null) {
+            int progress =
+                    (int)
+                            (progressInfo.getBytesDownloaded() * 100 / progressInfo.getTotalBytesToDownload());
+            // Set the progress for the progress bar.
+            progressBar.setProgress(progress);
+        }
+        // Handle failure status maybe…
+
+        // Unregister listener when there are no more install status updates.
+        if (isTerminateState(update.getInstallState())) {
+            moduleInstallClient.unregisterListener(this);
+        }
+    }
+
+    public boolean isTerminateState(@ModuleInstallStatusUpdate.InstallState int state) {
+        return state == STATE_CANCELED || state == STATE_COMPLETED || state == STATE_FAILED;
+    }
+}

--- a/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/ModuleInstallProgressListener.java
+++ b/packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/ModuleInstallProgressListener.java
@@ -4,30 +4,36 @@ import static com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpd
 import static com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate.InstallState.STATE_COMPLETED;
 import static com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate.InstallState.STATE_FAILED;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import com.google.android.gms.common.moduleinstall.InstallStatusListener;
+import com.google.android.gms.common.moduleinstall.ModuleInstall;
+import com.google.android.gms.common.moduleinstall.ModuleInstallClient;
 import com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate;
 
 public class ModuleInstallProgressListener implements InstallStatusListener {
+
+    @NonNull
+    private final BarcodeScanner implementation;
+
+    public ModuleInstallProgressListener(BarcodeScanner implementation) {
+        this.implementation = implementation;
+    }
+
+    public static boolean isTerminateState(@ModuleInstallStatusUpdate.InstallState int state) {
+        return state == STATE_CANCELED || state == STATE_COMPLETED || state == STATE_FAILED;
+    }
+
     @Override
     public void onInstallStatusUpdated(ModuleInstallStatusUpdate update) {
         ModuleInstallStatusUpdate.ProgressInfo progressInfo = update.getProgressInfo();
+        @ModuleInstallStatusUpdate.InstallState
+        int state = update.getInstallState();
         // Progress info is only set when modules are in the progress of downloading.
+        Integer progress = null;
         if (progressInfo != null) {
-            int progress =
-                    (int)
-                            (progressInfo.getBytesDownloaded() * 100 / progressInfo.getTotalBytesToDownload());
-            // Set the progress for the progress bar.
-            progressBar.setProgress(progress);
+            progress = (int) (progressInfo.getBytesDownloaded() / progressInfo.getTotalBytesToDownload());
         }
-        // Handle failure status maybe…
-
-        // Unregister listener when there are no more install status updates.
-        if (isTerminateState(update.getInstallState())) {
-            moduleInstallClient.unregisterListener(this);
-        }
-    }
-
-    public boolean isTerminateState(@ModuleInstallStatusUpdate.InstallState int state) {
-        return state == STATE_CANCELED || state == STATE_COMPLETED || state == STATE_FAILED;
+        implementation.handleGoogleBarcodeScannerModuleInstallProgress(state, progress);
     }
 }

--- a/packages/barcode-scanning/ios/Plugin/BarcodeScannerPlugin.m
+++ b/packages/barcode-scanning/ios/Plugin/BarcodeScannerPlugin.m
@@ -15,6 +15,8 @@ CAP_PLUGIN(BarcodeScannerPlugin, "BarcodeScanner",
            CAP_PLUGIN_METHOD(isTorchEnabled, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(isTorchAvailable, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(openSettings, CAPPluginReturnPromise);
+           CAP_PLUGIN_METHOD(isGoogleBarcodeScannerModuleAvailable, CAPPluginReturnPromise);
+           CAP_PLUGIN_METHOD(installGoogleBarcodeScannerModule, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(checkPermissions, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(requestPermissions, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(removeAllListeners, CAPPluginReturnPromise);

--- a/packages/barcode-scanning/ios/Plugin/BarcodeScannerPlugin.swift
+++ b/packages/barcode-scanning/ios/Plugin/BarcodeScannerPlugin.swift
@@ -164,6 +164,14 @@ public class BarcodeScannerPlugin: CAPPlugin {
             call.resolve()
         })
     }
+    
+    @objc func isGoogleBarcodeScannerModuleAvailable(_ call: CAPPluginCall) {
+        call.reject("Not available on iOS")
+    }
+    
+    @objc func installGoogleBarcodeScannerModule(_ call: CAPPluginCall) {
+        call.reject("Not available on iOS")
+    }
 
     @objc override public func checkPermissions(_ call: CAPPluginCall) {
         call.resolve([

--- a/packages/barcode-scanning/src/definitions.ts
+++ b/packages/barcode-scanning/src/definitions.ts
@@ -30,7 +30,10 @@ export interface BarcodeScannerPlugin {
   /**
    * Scan a barcode with a ready-to-use interface without WebView customization.
    *
-   * On **Android**, no camera permission is required.
+   * On **Android**, this method is only available on devices with Google Play Services
+   * installed. Therefore, no camera permission is required.
+   *
+   * **Attention:** Before using this method on *Android*, first check if the Google Barcode Scanner module is available.
    *
    * Only available on Android and iOS.
    *
@@ -95,6 +98,22 @@ export interface BarcodeScannerPlugin {
    */
   openSettings(): Promise<void>;
   /**
+   * Check if the Google Barcode Scanner module is available.
+   *
+   * Only available on Android.
+   *
+   * @since 5.1.0
+   */
+  isGoogleBarcodeScannerModuleAvailable(): Promise<IsGoogleBarcodeScannerModuleAvailableResult>;
+  /**
+   * Install the Google Barcode Scanner module.
+   *
+   * Only available on Android.
+   *
+   * @since 5.1.0
+   */
+  installGoogleBarcodeScannerModule(): Promise<void>;
+  /**
    * Check camera permission.
    *
    * Only available on Android and iOS.
@@ -131,6 +150,19 @@ export interface BarcodeScannerPlugin {
   addListener(
     eventName: 'scanError',
     listenerFunc: (event: ScanErrorEvent) => void,
+  ): Promise<PluginListenerHandle> & PluginListenerHandle;
+  /**
+   * Called when the Google Barcode Scanner module is installed.
+   *
+   * Available on Android.
+   *
+   * @since 5.1.0
+   */
+  addListener(
+    eventName: 'googleBarcodeScannerModuleInstallProgress',
+    listenerFunc: (
+      event: GoogleBarcodeScannerModuleInstallProgressEvent,
+    ) => void,
   ): Promise<PluginListenerHandle> & PluginListenerHandle;
   /**
    * Remove all listeners for this plugin.
@@ -255,6 +287,18 @@ export interface IsTorchAvailableResult {
 }
 
 /**
+ * @since 5.1.0
+ */
+export interface IsGoogleBarcodeScannerModuleAvailableResult {
+  /**
+   * Whether or not the Google Barcode Scanner module is available.
+   *
+   * @since 5.1.0
+   */
+  available: boolean;
+}
+
+/**
  * @since 0.0.1
  */
 export type CameraPermissionState = PermissionState | 'limited';
@@ -291,6 +335,24 @@ export interface ScanErrorEvent {
    * @since 0.0.1
    */
   message: string;
+}
+
+/**
+ * @since 5.1.0
+ */
+export interface GoogleBarcodeScannerModuleInstallProgressEvent {
+  /**
+   * The current state of the installation.
+   *
+   * @since 5.1.0
+   */
+  state: GoogleBarcodeScannerModuleInstallState;
+  /**
+   * The progress of the installation in percent between 0 and 100.
+   *
+   * @since 5.1.0
+   */
+  progress?: number;
 }
 
 /**
@@ -504,4 +566,42 @@ export enum LensFacing {
    * @since 0.0.1
    */
   Back = 'BACK',
+}
+
+/**
+ * @since 5.1.0
+ */
+export enum GoogleBarcodeScannerModuleInstallState {
+  /**
+   * @since 5.1.0
+   */
+  UNKNOWN = 0,
+  /**
+   * @since 5.1.0
+   */
+  PENDING = 1,
+  /**
+   * @since 5.1.0
+   */
+  DOWNLOADING = 2,
+  /**
+   * @since 5.1.0
+   */
+  CANCELED = 3,
+  /**
+   * @since 5.1.0
+   */
+  COMPLETED = 4,
+  /**
+   * @since 5.1.0
+   */
+  FAILED = 5,
+  /**
+   * @since 5.1.0
+   */
+  INSTALLING = 6,
+  /**
+   * @since 5.1.0
+   */
+  DOWNLOAD_PAUSED = 7,
 }

--- a/packages/barcode-scanning/src/definitions.ts
+++ b/packages/barcode-scanning/src/definitions.ts
@@ -108,6 +108,10 @@ export interface BarcodeScannerPlugin {
   /**
    * Install the Google Barcode Scanner module.
    *
+   * **Attention**: This only starts the installation.
+   * The `googleBarcodeScannerModuleInstallProgress` event listener will
+   * notify you when the installation is complete.
+   *
    * Only available on Android.
    *
    * @since 5.1.0

--- a/packages/barcode-scanning/src/web.ts
+++ b/packages/barcode-scanning/src/web.ts
@@ -2,6 +2,7 @@ import { CapacitorException, ExceptionCode, WebPlugin } from '@capacitor/core';
 
 import type {
   BarcodeScannerPlugin,
+  IsGoogleBarcodeScannerModuleAvailableResult,
   IsSupportedResult,
   IsTorchAvailableResult,
   IsTorchEnabledResult,
@@ -59,6 +60,14 @@ export class BarcodeScannerWeb
   }
 
   async openSettings(): Promise<void> {
+    throw this.createUnavailableException();
+  }
+
+  async isGoogleBarcodeScannerModuleAvailable(): Promise<IsGoogleBarcodeScannerModuleAvailableResult> {
+    throw this.createUnavailableException();
+  }
+
+  async installGoogleBarcodeScannerModule(): Promise<void> {
     throw this.createUnavailableException();
   }
 

